blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AbstractMetricsRW.java

```diff
@@ -52,8 +53,16 @@
             CacheBuilder.newBuilder().expireAfterAccess(10,
                         TimeUnit.MINUTES).concurrencyLevel(16).build();
 
+    // this collection is used to reduce the number of delayed locators that get written per slot. Simply, if a locator
+    // has been seen for a slot, don't bother.
+    private static final Cache<String, Boolean> insertedDelayedLocators = CacheBuilder.newBuilder().expireAfterAccess(10,
```

Comment:
We already have the cache for metrics locator in two places, one for astyanax and one for datastax. I added one more for delayed locators.